### ISSUE TITLE
Event retention // delete events after a certain timeframe

### ISSUE THREAD

**Issue by MarconLP** [PostHog/posthog#17031]
## Is your feature request related to a problem?

Some customers would like to delete their events after a certain amount of time (e.g: 2 years)

## Describe the solution you'd like

Add the option to delete events after a certain timeframe

## Additional context

related tickets:
- https://posthoghelp.zendesk.com/agent/tickets/4267
- https://posthoghelp.zendesk.com/agent/tickets/5045

**Comment by nosovk** [comment]
It would be nice to have that feature. Actually we have to keep 1 year of data now, but we actually don't need it. It would be nice to store last month and all. Actually we could resetap posthog instance every month, but it's a kreepy solution comparing to regulated retention time.

**Comment by johnbizokk** [comment]
Hello! Is it true that the self-hosted version has no retention settings regarding the events? I have found some settings related to the recording retention - **RECORDINGS_TTL_WEEKS** and **RECORDINGS_PERFORMANCE_EVENTS_TTL_WEEKS**, but there are no settings for the events. If so, it would be nice to have dedicated settings for event retention for the micro-installation (16 GB SD card), such as a Raspberry Pi.

**Comment by o-m-chumak** [comment]
It will be great to have the option to automatically delete events after a certain timeframe. This feature would provide users with more control over data retention and align with various privacy and data management requirements. Looking forward to seeing this enhancement in future releases.

**Comment by MarconLP** [comment]
+1 https://posthoghelp.zendesk.com/agent/tickets/8811 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/11254)

**Comment by MarconLP** [comment]
+1 https://posthoghelp.zendesk.com/agent/tickets/9651 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/12106)

**Comment by tricky42** [comment]
Hi, 

we also need a feature for Posthog EU to define the data retention of events. It would be great to get official feedback on this topic.

**Comment by lezgin-bakircioglu-qred** [comment]
> Hi,
> 
> we also need a feature for Posthog EU to define the data retention of events. It would be great to get official feedback on this topic.

Same here

**Comment by johnbizokk** [comment]
Hello! Is there any progress on this issue?

**Comment by MarconLP** [comment]
+1 https://posthoghelp.zendesk.com/agent/tickets/19641 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/22102)

**Comment by luke-belton** [comment]
+1 [#32941](https://posthoghelp.zendesk.com/agent/tickets/32941) (moved to PostHog: https://us.posthog.com/project/2/support/tickets/34616)

**Comment by ftlno** [comment]
+1

**Comment by Wuemeli** [comment]
+1

**Comment by chandansinghrebtel** [comment]
+1

**Comment by maurice-g** [comment]
+1

**Comment by rnflt** [comment]
+1

**Comment by Saboor-Hakimi** [comment]
This issue has been open for over two years.
There should have been some progress by now.

**Comment by HaynesPostHog** [comment]
+1 https://posthoghelp.zendesk.com/agent/tickets/39420 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/41596)

**Comment by snickn** [comment]
+1 being able to delete events after X years would be helpful for align to our data retention policies

**Comment by phillram** [comment]
+1 https://posthoghelp.zendesk.com/agent/tickets/54296 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/56370)

**Comment by darkopia** [comment]
+1 https://posthoghelp.zendesk.com/agent/tickets/57120 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/58849)

**Comment by ajs888GB** [comment]
+1

**Comment by keithhackbarth** [comment]
+1 - This is a blocker for enterprise deployment of Posthog

**Comment by hongweixLi** [comment]
+1  any updates here ?

**Comment by simonvb-xbow** [comment]
+1

**Comment by listen3000** [comment]
We can't justify keeping the data for a year while we need user attribution. Blocked by this.

**Comment by HaynesPostHog** [comment]
+1 https://posthoghelp.zendesk.com/agent/tickets/59879 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/61503)

**Comment by max-perry** [comment]
+1

**Comment by Moarcoo** [comment]
+1 please implement this. For GDPR reasons this is **very** important

**Comment by martincampagnoli** [comment]
+1

**Comment by mawywlf** [comment]
+1

**Comment by Mittchel** [comment]
+1 please implement this

**Comment by MarcoKamphuis1** [comment]
+1

**Comment by hpouillot** [comment]
+1 https://posthoghelp.zendesk.com/agent/tickets/61083 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/62423)

**Comment by atroianof** [comment]
Same requirement here

➕ 1️⃣

**Comment by phillram** [comment]
+1 https://posthoghelp.zendesk.com/agent/tickets/61470 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/61757)

**Comment by nosovk** [comment]
The issue is 3 years lasting. It seems that it's just not bother core team, because most of setups are launch and forget. This feature required only by long lasting environments.

**Comment by luke-belton** [comment]
+1 [#62148](https://posthoghelp.zendesk.com/agent/tickets/62148) (moved to PostHog: https://us.posthog.com/project/2/support/tickets/62717)

**Comment by phillram** [comment]
+1 https://posthoghelp.zendesk.com/agent/tickets/62631 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/63354)

**Comment by darkopia** [comment]
+1 https://us.posthog.com/project/2/support/tickets/64686

**Comment by phillram** [comment]
+1 https://us.posthog.com/project/2/support/tickets/65221

**Comment by phillram** [comment]
+1 https://us.posthog.com/project/2/support/tickets/65204

**Comment by xljones** [comment]
+1 https://us.posthog.com/project/2/support/tickets/65175

**Comment by HaynesPostHog** [comment]
+1 https://us.posthog.com/project/2/support/tickets/65593

**Comment by phillram** [comment]
+1 https://us.posthog.com/project/2/support/tickets/65661

**Comment by phillram** [comment]
+1 https://us.posthog.com/project/2/support/tickets/65664

**Comment by phillram** [comment]
+1 https://us.posthog.com/project/2/support/tickets/66116

**Comment by dianaprior** [comment]
+1 from Prior Labs (https://priorlabs.ai/) (I am Staff PM responsible for our analytics setup) - very painful that Posthog doesn't have it, at least on the cloud version. We really need this for compliance with GDPR and co.

**Comment by luke-belton** [comment]
+1 [#66193](https://us.posthog.com/project/2/support/tickets/66193?view=yBA4lSxy)

**Comment by darkopia** [comment]
+1 65044

**Comment by bportnoy** [comment]
+1

**Comment by phillram** [comment]
+1 https://us.posthog.com/project/2/support/tickets/66728
+1 https://us.posthog.com/project/2/support/tickets/66762
+1 https://us.posthog.com/project/2/support/tickets/66502

**Comment by darkopia** [comment]
+1 https://us.posthog.com/project/2/support/tickets/67121

**Comment by HaynesPostHog** [comment]
+1 https://us.posthog.com/project/2/support/tickets/67111

**Comment by thejozika** [comment]
+1

**Comment by darkopia** [comment]
+1 67625

**Comment by xljones** [comment]
+1 https://us.posthog.com/project/2/support/tickets/67957 (per-project retention adjustment)

**Comment by phillram** [comment]
+1 https://us.posthog.com/project/2/support/tickets/68175
+1 https://us.posthog.com/project/2/support/tickets/68424

**Comment by xljones** [comment]
+1 ticket 68440

**Comment by phillram** [comment]
+1 https://us.posthog.com/project/2/support/tickets/69001

**Comment by darkopia** [comment]
+1 69137

**Comment by CShine612** [comment]
+1 69188

**Comment by darkopia** [comment]
+1 69634

**Comment by alinph** [comment]
+1 69974

**Comment by HaynesPostHog** [comment]
+2 https://us.posthog.com/project/2/support/tickets/70176, https://us.posthog.com/project/2/support/tickets/70184

**Comment by darkopia** [comment]
+1 70382

**Comment by xljones** [comment]
+1 70457

**Comment by xljones** [comment]
+1 71239

**Comment by xljones** [comment]
+1 71389

**Comment by CShine612** [comment]
+1 71513

**Comment by HaynesPostHog** [comment]
+1 71865

**Comment by darkopia** [comment]
+1 71963

**Comment by alinph** [comment]
+1 72212

**Comment by CShine612** [comment]
+1 71853
